Mathlib/Analysis/InnerProductSpace/Projection.lean

```diff
@@ -552,6 +591,13 @@ theorem orthogonalProjection_eq_zero_iff {v : E} : K.orthogonalProjection v = 0
 theorem ker_orthogonalProjection : LinearMap.ker K.orthogonalProjection = Kᗮ := by
   ext; exact orthogonalProjection_eq_zero_iff
 
+@[simp]
+lemma ker_starProjection (U : Submodule 𝕜 E) [U.HasOrthogonalProjection] :
+    LinearMap.ker U.starProjection = Uᗮ := by
+  ext x
+  simp only [starProjection, LinearMap.mem_ker, ContinuousLinearMap.coe_comp', coe_subtypeL',
+    coe_subtype, Function.comp_apply, ZeroMemClass.coe_eq_zero, orthogonalProjection_eq_zero_iff]
+
```

Comment:
If you move the results at the end of this file like I suggest, then you can do:
```suggestion
open ContinuousLinearMap in
@[simp]
lemma ker_starProjection (U : Submodule 𝕜 E) [U.HasOrthogonalProjection] :
    LinearMap.ker U.starProjection = Uᗮ := by
  rw [(isIdempotentElem_starProjection U).ker_eq_range, ← starProjection_orthogonal',
    range_starProjection]
```